### ISSUE TITLE
Detailed parameter setting for training DOTAv1 in the OBB task

### ISSUE THREAD

**Issue by Fusica** [ultralytics/ultralytics#7578]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I tried to train yolov8n-obb.yaml for 200 epochs on my own, and the results were generally reproducible, but in the end I found that there was a difference of about 14% in AP50. So I'm curious about what parameter settings were used for the pre-training to obtain the experimental results. This would be helpful for us to reproduce the results and use more targeted parameters to train our own rotated object detection dataset. Thank you very much!
<img width="818" alt="image" src="https://github.com/ultralytics/ultralytics/assets/33285698/27a6c7dc-4c8a-4ad9-8b98-7f809605a19f">

**Comment by glenn-jocher** [comment]
@Fusica hello! Thanks for reaching out with your question regarding the training of the YOLOv8 model on the DOTAv1 dataset for the OBB task.

For training YOLOv8 on rotated object detection datasets like DOTAv1, it's crucial to ensure that the dataset is correctly formatted and that the model configuration, including anchors and hyperparameters, is appropriately set for the task.

The pre-training settings used for our experiments are detailed in the model's YAML configuration file. This file includes the model architecture, data pipeline settings, and hyperparameters such as learning rate, weight decay, and augmentation settings. To closely reproduce our results, you should use the same configuration file and ensure that your dataset is prepared similarly to how the pre-training dataset was prepared.

For the OBB task, special attention should be given to the anchor settings, as they can significantly impact the performance of rotated object detection. Make sure to adjust the anchors to better fit the aspect ratios and scales of the objects in your dataset.

If you're experiencing a 14% difference in AP50, it could be due to several factors, including differences in the training data, initialization, or even slight variations in the training process. It's also worth reviewing the validation process to ensure it aligns with the pre-training evaluation.

For more detailed guidance on training, hyperparameter tuning, and dataset preparation, please refer to our documentation at https://docs.ultralytics.com. The documentation provides comprehensive instructions that can help you fine-tune the training process for your specific use case.

Keep experimenting, and don't hesitate to reach out if you have further questions. Good luck with your training! 🚀

**Comment by Fusica** [comment]
Thank you very much for your response, but I'm sorry to say that I didn't find the config file specifically related to the OBB task. I only saw default.yaml and don't know if this is the one. 
<img width="227" alt="image" src="https://github.com/ultralytics/ultralytics/assets/33285698/eea19f82-3fb5-473f-b978-c9c395cacb38">
Also, when using torch.load to check yolov8n.pt, I found that the pretrained parameter was set to True. Can you confirm whether COCO's pre-trained weights were loaded during your training?
<img width="183" alt="image" src="https://github.com/ultralytics/ultralytics/assets/33285698/6e9e75c0-de4c-44ed-bb2f-ba1770051ff6">

**Comment by glenn-jocher** [comment]
@Fusica apologies for the confusion. The `default.yaml` is a general configuration file. For the OBB task, you should create a custom YAML file tailored to the specifics of your dataset and task, including the appropriate anchors and hyperparameters.

Regarding the pre-trained weights, yes, the `yolov8n.pt` file typically includes COCO pre-trained weights by default, which are used to initialize the network for further training on a specific task like OBB. This helps in achieving better performance, especially when your dataset is similar to COCO in terms of object categories and scales.

**Comment by puppetmonkey** [comment]
楼主复现出来吗？最后使用哪一种超参数配置文件？

**Comment by Fusica** [comment]
@puppetmonkey I am trying to use the pre-trained `yolov8n.pt` for 200 epoch of training, still using `default.yaml`. I haven't got any results yet. If it can be reproduced, I will close this issue.

**Comment by kuazhangxiaoai** [comment]
我弄了一个OBB的项目，可以看一下：https://github.com/kuazhangxiaoai/yolov8-obb.git
![yolov8-obb](https://github.com/ultralytics/ultralytics/assets/54108453/0f776110-0e01-4f84-9fec-1db4149b4f36)

**Comment by glenn-jocher** [comment]
@kuazhangxiaoai, it's great to see community projects like yours! For any specific questions or discussions about your OBB project, I encourage you to use the Discussions section of our repo. This helps keep the Issues section focused on direct support and bug tracking for YOLOv8. Keep up the good work! 🌟

**Comment by Fusica** [comment]
@kuazhangxiaoai May I ask which dataset was used to train the model that produced this single image result?

**Comment by kuazhangxiaoai** [comment]
@Fusica You can use the pretrained model from https://docs.ultralytics.com/tasks/obb/ or train on DOTA dataset by yourself.

**Comment by puppetmonkey** [comment]
@kuazhangxiaoai 请问 图片结果，是怎么获得的？ 用了哪个模型？使用了哪个预训练权重？哪个数据集？  
![image](https://github.com/ultralytics/ultralytics/assets/81809745/6ebf1dd6-14a8-45e8-abc4-a58fb55ecde3)

**Comment by puppetmonkey** [comment]
@Fusica 大佬复现的结果怎么样？

**Comment by kuazhangxiaoai** [comment]
@puppetmonkey 使用https://github.com/ultralytics/assets/releases/download/v8.1.0/yolov8n-obb.pt，运行obb/val.py得到的结果。训练的代码预计周五前更新，请关注https://github.com/kuazhangxiaoai/yolov8-obb.git

**Comment by Helloworld777777** [comment]
I had the same problem.May I ask if you have solved it now?If possible,I hope to get your help.Thank you very much. @Fusica @puppetmonkey @Burhan-Q

**Comment by Burhan-Q** [comment]
```py
from pprint import pprint as pp
from ultralytics import YOLO

model = YOLO("yolov8n-obb.pt")

pp(model.ckpt.get("train_args"))
```

**Comment by tuilala102** [comment]
I am interested in this area, can you tell me what improvements you have made compared to the standard yolov8-obb ​​model to get such training results?

**Comment by tuilala102** [comment]
> [kuazhangxiaoai](/kuazhangxiaoai)
I am interested in this area, can you tell me what improvements you have made compared to the standard yolov8-obb ​​model to get such training results?
